agent/agent.go

```diff
@@ -795,6 +803,8 @@ func (a *agent) handleReconnectingPTY(ctx context.Context, msg codersdk.Reconnec
 					// When the PTY is closed, this is triggered.
 					break
 				}
+
+				lastRead = time.Now()
```

Comment:
Assignment/read to this variable is racy, perhaps we could put it behind the mutex we already have for the rpty circbuf.